src/main/java/duke/TaskArrayList.java

```diff
@@ -0,0 +1,106 @@
+package duke;
+
+import duke.exceptions.DukeException;
+import duke.tasks.Task;
+
+import java.util.ArrayList;
+
+/**
+ * Array List of Task objects.
+ */
+public class TaskArrayList extends ArrayList<Task> {
+    public final static String DELETE_USAGE_TEXT = "Usage: delete <integer task number>";
+    public final static String DONE_USAGE_TEXT = "Usage: done <integer task number>";
+    public final static String FIND_USAGE_TEXT = "Usage: find <search String>";
+
+    public TaskArrayList() {
+        super();
+    }
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param task Task object to add.
+     * @return addition message String.
+     */
+    public String addTask(Task task) {
+        this.add(task);
+        return task.addMsg() + "\n" + this.announceNewLength();
+    }
+
+    /**
+     * Marks the given task as done.
+     *
+     * @param taskNo task number to mark as done.
+     * @return removal message String.
```

Comment:
I believe the first letter of the descriptions for the param and return should be capitalized!